cmd/xl-v1-multipart.go

```diff
@@ -416,6 +416,14 @@ func (xl xlObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	// Delete the temporary object part. If PutObjectPart succeeds there would be nothing to delete.
 	defer xl.deleteObject(minioMetaBucket, tmpPartPath)
 
+	if size > 0 {
+		for _, disk := range onlineDisks {
+			if disk != nil {
+				_ = disk.PrepareFile(minioMetaBucket, tmpPartPath, size/2+size%2)
+			}
```

Comment:
> Do you need to know exact padding ?

we can put any number > 0 here, however we will lose the benefit of fallocate if we put a number inferior to  `size/2+size%2`

By the way, I need to see if I can move this to a more low level function which knows in advance how to calculate the size of output files (data + parity)